src/cpp/rtps/participant/RTPSParticipantImpl.cpp

```diff
@@ -2138,6 +2168,26 @@ DurabilityKind_t RTPSParticipantImpl::get_persistence_durability_red_line(
     return durability_red_line;
 }
 
+void RTPSParticipantImpl::environment_file_has_changed()
+{
+    RTPSParticipantAttributes patt;
```

Comment:
```suggestion
    RTPSParticipantAttributes patt = m_att;
```
Otherwise, we may be inadvertently changing userData.